src/main/java/org/zalando/nakadi/repository/TopicRepositoryHolder.java

```diff
@@ -27,40 +30,57 @@
     private static final Logger LOG = LoggerFactory.getLogger(TopicRepositoryHolder.class);
 
     private final Map<Storage.Type, TopicRepositoryCreator> repositoryCreators;
-    private final ReadWriteLock readWriteLock;
     private final Map<Storage, TopicRepository> storageTopicRepository;
 
+    private final Lock lock = new ReentrantLock();
```

Comment:
why have you refactored locking?

Reply:
To allow getting topic repositories in parallel while it is being created